code/modules/library/lib_machines.dm

```diff
@@ -449,25 +465,50 @@ datum/borrowbook // Datum used to keep track of who has borrowed what when and f
  * Book binder
  */
 /obj/machinery/bookbinder
-	name = "Book Binder"
+	name = "book binder"
+	desc = "A machine that takes paper and binds them into books. Fascinating!"
 	icon = 'icons/obj/library.dmi'
 	icon_state = "binder"
-	anchored = 1
-	density = 1
+	anchored = TRUE
+	density = TRUE
+	var/binding = FALSE
 
-/obj/machinery/bookbinder/attackby(var/obj/O as obj, var/mob/user as mob)
+/obj/machinery/bookbinder/attackby(var/obj/O, mob/user)
 	if(istype(O, /obj/item/paper))
+		if(!anchored)
+			to_chat(user, SPAN_WARNING("\The [src] must be secured to the floor first!"))
+			return
+		if(binding)
+			to_chat(user, SPAN_WARNING("You must wait for \the [src] to finish its current operation!"))
+			return
+		var/turf/T = get_turf(src)
 		user.drop_from_inventory(O,src)
-		user.visible_message("[user] loads some paper into [src].", "You load some paper into [src].")
-		src.visible_message("[src] begins to hum as it warms up its printing drums.")
-		playsound(src.loc, 'sound/bureaucracy/binder.ogg', 75, 1)
+		user.visible_message(SPAN_NOTICE("\The [user] loads some paper into \the [src]."), SPAN_NOTICE("You load some paper into \the [src]."))
+		visible_message(SPAN_NOTICE("\The [src] begins to hum as it warms up its printing drums."))
+		playsound(T, 'sound/bureaucracy/binder.ogg', 75, 1)
+		binding = TRUE
 		sleep(rand(200,400))
-		src.visible_message("[src] whirs as it prints and binds a new book.")
-		playsound(src.loc, 'sound/bureaucracy/print.ogg', 75, 1)
-		var/obj/item/book/b = new(src.loc)
+		binding = FALSE
+		if(!anchored)
+			visible_message(SPAN_WARNING("\The [src] buzzes and flashes an error light."))
+			O.forceMove(T)
+			return
+		visible_message(SPAN_NOTICE("\The [src] whirs as it prints and binds a new book."))
+		playsound(T, 'sound/bureaucracy/print.ogg', 75, 1)
+		var/obj/item/book/b = new(T)
 		b.dat = O:info
 		b.name = "Print Job #" + "[rand(100, 999)]"
 		b.icon_state = "book[rand(1,7)]"
 		qdel(O)
-	else
-		..()
+		return
+	if(O.iswrench())
+		playsound(get_turf(src), O.usesound, 75, TRUE)
```

Comment:
`use_tool` instead?

Reply:
`use_tool` is for processes that have a delay, such as welding, or things that involve a `do_after`. Instant actions are not currently supported by `use_tool` as it returns if no delay is set.